From c84e20ab2c1459c80106f776b7ba60b768ae88ff Mon Sep 17 00:00:00 2001
From: Victor Gu <xigu@marvell.com>
Date: Fri, 24 Jun 2016 21:54:41 +0800
Subject: [PATCH 0529/2241] spi: a3700: update NOR SPI flash partition

Update the NOR SPI flash partition definition.
- enlarge bootloader size from 1M to 2M bytes
- add "U-boot Env" partion for u-boot env variables
- update the shift of Linux partition and enlarge
  the size to fully use the 16M NOR flash in DB board.
- JIRA SYSTEMSW-2370
Change-Id: I33d3c014b05773457b805d83cb46fec56626e90d
Signed-off-by: Victor Gu <xigu@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/30710
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-3720-db.dts | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db.dts b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
index 224c2b3..986be13 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
@@ -77,12 +77,16 @@
 					pinctrl-0 = <&spi0_pins &spi_quad_pins>;
 
 					partition@0 {
-						label = "U-boot";
-						reg = <0 0x100000>;
+						label = "Bootloader";
+						reg = <0 0x200000>;
 					};
-					partition@100000 {
+					partition@200000 {
+						label = "U-boot Env";
+						reg = <0x200000 0x10000>;
+					};
+					partition@210000 {
 						label = "Linux";
-						reg = <0x100000 0x300000>;
+						reg = <0x210000 0xDF0000>;
 					};
 				};
 			};
-- 
2.7.4

